lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -374,12 +375,17 @@ class BatchEnrollment
 
     # attach click handler for enrollment buttons
     @$enrollment_button.click (event) =>
+      if not @$reason_field.val()
+        @fail_with_error gettext "Reason field should not be left blank."
+        return false
+
```

Comment:
As mentioned below, I think only paywalled courses should have a mandatory reason filled in